space/core/components/issues/issue-layouts/with-display-properties-HOC.tsx

```diff
@@ -0,0 +1,26 @@
+import { ReactNode } from "react";
+import { observer } from "mobx-react";
+import { IIssueDisplayProperties } from "@plane/types";
+
+interface IWithDisplayPropertiesHOC {
+  displayProperties: IIssueDisplayProperties;
+  shouldRenderProperty?: (displayProperties: IIssueDisplayProperties) => boolean;
+  displayPropertyKey: keyof IIssueDisplayProperties | (keyof IIssueDisplayProperties)[];
+  children: ReactNode;
+}
+
+export const WithDisplayPropertiesHOC = observer(
```

Comment:
Wouldn't it be better if its a function instead of a HOC component? There is no fallback or a loader.

Reply:
HOC i feel would be better, because it avoids adding the logic to the children which can be used independently else where.

Reply:
I'm suggesting to use a function to conditionally render. Your function will take the display properties and the current key and return a boolean. Exactly what your HOC is doing, except as a function.